File: python/test/t_LHSExperiment_std.py
#! /usr/bin/env python

import openturns as ot

ot.TESTPREAMBLE()

distribution = ot.Normal(4)
size = 10
experiment = ot.LHSExperiment(distribution, size)
print("experiment = ", experiment)
sample, weights = experiment.generateWithWeights()
print("sample = ", repr(sample))
print("weights = ", repr(weights))
print("sample2 = ", experiment.generate())
experiment = ot.LHSExperiment(distribution, size, True, True)
print("sample  = ", experiment.generate())
print("sample2 = ", experiment.generate())
experiment = ot.LHSExperiment(distribution, size, False, False)
print("sample  = ", experiment.generate())
print("sample2 = ", experiment.generate())
experiment = ot.LHSExperiment(distribution, size, True, False)
print("sample  = ", experiment.generate())
print("sample2 = ", experiment.generate())
experiment = ot.LHSExperiment(size)
print("sample  = ", experiment.generate())
print("sample2 = ", experiment.generate())

# variable size
experiment = ot.LHSExperiment(ot.Normal(4), 10)
for size in [215, 464]:
    experiment.setSize(size)
    X = experiment.generate()

# with correlation
dimension = 3
mu = ot.Point(dimension)
sigma = ot.CovarianceMatrix(dimension)
sigma[0, 2] = -0.5
distribution = ot.Normal(mu, sigma)
sampleSize = 5
experiment = ot.LHSExperiment(distribution, sampleSize)
sample = experiment.generate()

File: lib/test/t_LHSExperiment_std.cxx
//                                               -*- C++ -*-
/**
 *  @brief The test file of class LHSExperiment for standard methods
 *
 *  Copyright 2005-2016 Airbus-EDF-IMACS-Phimeca
 *
 *  This library is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Lesser General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
#include "openturns/OT.hxx"
#include "openturns/OTtestcode.hxx"

using namespace OT;
using namespace OT::Test;

int main(int argc, char *argv[])
{
  TESTPREAMBLE;
  OStream fullprint(std::cout);
  setRandomGenerator();

  try
  {
    Normal distribution(4);
    UnsignedInteger size(10);
    LHSExperiment myPlane(distribution, size);
    fullprint << "myPlane = " << myPlane << std::endl;
    NumericalPoint weights(0);
    NumericalSample sample(myPlane.generateWithWeights(weights));
    fullprint << "sample  = " << sample << std::endl;
    fullprint << "weights = " << weights << std::endl;
    fullprint << "sample2 = " << myPlane.generate() << std::endl;
    myPlane = LHSExperiment(distribution, size, true, true);
    fullprint << "sample  = " << myPlane.generate() << std::endl;
    fullprint << "sample2 = " << myPlane.generate() << std::endl;
    myPlane = LHSExperiment(distribution, size, false, false);
    fullprint << "sample  = " << myPlane.generate() << std::endl;
    fullprint << "sample2 = " << myPlane.generate() << std::endl;
    myPlane = LHSExperiment(distribution, size, true, false);
    fullprint << "sample  = " << myPlane.generate() << std::endl;
    fullprint << "sample2 = " << myPlane.generate() << std::endl;
  }
  catch (TestFailed & ex)
  {
    std::cerr << ex << std::endl;
    return ExitCode::Error;
  }

  return ExitCode::Success;
}
